### ISSUE TITLE
Riley Mete: 1-1 

### ISSUE THREAD

**Issue by rmete** [CSC307-W2020-Team6/307-Sec01-Team6#29]
# Am I Blocked?

- I currently am not blocked; however, I believe there may be some roadblocks when I try to learn how to use Django, the Python Web Framework.

# What I Did

- Brainstormed with my team and on and began created issues and projects 
- I was responsible for creating the following issues: #17, #27, #22    
- Became more comfortable with GitHub and began familiarizing myself with markdown
- As a team, we decided on three features for our MVP. The three features we decided on are documented in the following issues: #9, #17, #22  
- I was assigned to work on back-end development. To do this, I will be using the [Python](https://www.python.org/) programming language, and I plan to utilize [Django](https://www.djangoproject.com/) as the framework.  

# What I Plan to do Next Week

- I plan on familiarizing myself with [Django](https://www.djangoproject.com/) and beginning the back-end of our project.
- I plan to work on my assigned feature: Adding Friends #17

**Comment by rmete** [comment]
# Am I Blocked?

- I currently am blocked due to issues with our database. When doing the Django tutorial, I ran into a part that required me to set up a database; however, the suggested databases were SQLite, PostgreSQL, MySQL, and Oracle. Originally we were going to use Firebase, however, after some further research, we found out that this database would greatly limit the functionality of Django. Because of this, we had to begin using a new database, MySQL, but were are having trouble downloading and getting it to work. 


# What I Did

- Updated the User Story, Biz Goal, and Use Case of issue #27 
- I downloaded [Django](https://www.djangoproject.com/) and began the installation process.
- I also downloaded and installed [PyCharm](https://www.jetbrains.com/pycharm/), as I wanted to use an IDE for this project. 
- I began to familiarize myself with both PyCharm and Django. I then began the coding process for our product LyncUp, which is where I ran into the database issue mentioned above.
- I was able to get a working URL for our webpage; however, this is where my progress was halted. Our current web page, which at this point is just a blank, Django-defaulted shell, is currently parked at this [URL](http://127.0.0.1:8000/)
- I pushed the code I created in PyCharm so that my team members can view what I have done. The code I pushed all resides within the [LyncUp](https://github.com/MJaojoco/307-Sec01-Team6/tree/master/LyncUp) directory. 
# What I Plan to do Next Week

- This week I plan on sitting down and working with Gabe, our database developer. My goal is to finish the week one deliverables that we were unable to accomplish, as well as get a good start on the backend of our product. 
- I plan on trying to set up a Google calendar for our team so we can be more efficient with our scheduling.

**Comment by rmete** [comment]
After talking to Professor Ford, I found out that SQLite is the ideal choice for our database as it is built into Python. With this new information, I believe I will now be unblocked and can continue with my work!

**Comment by rmete** [comment]
# Am I Blocked?

- I currently am blocked due to the fact that the database is currently on our teammate's local machine. We will try to figure this issue out first so we can proceed with our other issues. 

# What I Did

- I worked with my team to figure out a new ideal stack. We are now using SQLite and React. 
- Help restructure our codebase by putting the database code into a database folder and putting my code written with Django in a Django folder. 
- I researched into Django's calendar packages as well as Google calendar integration. I am going to play around with both to see which one I prefer.
- Helped decide which items we as a team wanted in our sprint 2 project boards
# What I Plan to do Next Week

- This week I plan to figure out which calendar I want to use for our project: Google Calendar or Django-Schedule.
- I plan to complete issue number #11 as well as begin issue #14. Ideally, I want to complete both by Sunday.

**Comment by saraford** [comment]
> to the fact that the database is currently on our teammate's local machine.

I've told other teams to have each team member have a local copy of the database running on each laptop, just to keep things simple. You can say that one laptop is the "production" laptop that you'll give me to demo on the last day of class. HTH.

**Comment by rmete** [comment]
# Am I Blocked?

- No, currently I am not blocked; however, our team found out that the current stack we were using was not going to work

# What I Did

- As a team, we decided to use Django for only framework
- This week I worked with the Goggle Calendar API and I am currently still working on importing it into python. 
- I wrote a test case to cover whether or not a user could connect to the database. This test case can be viewed in the [Unit Test](https://github.com/MJaojoco/307-Sec01-Team6/issues/36) issue
- I was in charge of deciding which tests we would use for our [Regression Testing](https://github.com/MJaojoco/307-Sec01-Team6/issues/37) for the [Creating Users](https://github.com/MJaojoco/307-Sec01-Team6/issues/26) feature 


# What I plan to do

- I plan on fishing up with integrating and authenticating the Google Calendar API.
- I plan on figuring out how to interact with the API and our database

**Comment by rmete** [comment]
# Am I Blocked?

- No, currently I am not blocked

# What I Did

- This week our team utilized Django to create a working site.
- We added account creating to our site, and users can now log in and have their credentials saved
- Began to familiarize myself with HTML5 as well as with Bootstrap



# What I plan to do

- Next week we plan on adding more functionality to our site. In addition, I plan on getting more comfortable with HTML5 so I can assist with front end development.

**Comment by rmete** [comment]
# Am I Blocked?

- No, currently I am not blocked

# What I Did

- Further familiarized myself with Bootsrapt.
- Learned how to use selenium; however, I experienced an error relating to launching our website 



# What I plan to do

- This week we plan on adding more functionality and hopefully finishing our application by Friday.

**Comment by saraford** [comment]
I need to see a link to the Work Items (or Spikes) for "What I did" in order to receive credit for this lab. Please update and then send me email when you want me to regrade.

**Comment by rmete** [comment]
# Am I Blocked?

- No, currently I am not blocked

# What I Did

- Created test cases to test Post creation for our app
- Created a .travis.yml file to download our dependencies and run test cases for our project. This allows us to have continuoius integration #63
- I started implementing different scheduling plug-ins; however, I have yet to commit the code as we have not found one we are completely content with. 



# What I plan to do

- This week we plan on adding more functionality and getting groups and scheduling completely squared away
